src/System.Private.ServiceModel/tests/Scenarios/Binding/Http/NetHttpsBindingTests.cs

```diff
@@ -0,0 +1,116 @@
+
+
+using System;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+public static class Binding_Http_NetHttpsBindingTests
+{
+    [Fact]
+    [OuterLoop]
+    public static void StandardUsage_NetHttps_Echo_RoundTrips_String()
+    {
+        string testString = "Hello";
+        ChannelFactory<IWcfService> factory = null;
+        IWcfService serviceProxy = null;
+
+        try
+        {
+            // *** SETUP *** \\
+            NetHttpsBinding netHttpsBinding = new NetHttpsBinding();
+            factory = new ChannelFactory<IWcfService>(netHttpsBinding, new EndpointAddress(Endpoints.HttpBaseAddress_NetHttps));
+            serviceProxy = factory.CreateChannel();
+
+            // *** EXECUTE *** \\
+            string result = serviceProxy.Echo(testString);
+
+            // *** VALIDATE *** \\
+            Assert.NotNull(result);
+            Assert.Equal(testString, result);
+
+            // *** CLEANUP *** \\
+            factory.Close();
+            ((ICommunicationObject)serviceProxy).Close();
+        }
+        finally
+        {
+            // *** ENSURE CLEANUP *** \\
+            ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
+        }
+    }
+
+    [Fact]
+    [OuterLoop]
+    public static void TransportWithMessageCredential_NotSupported_NetHttps()
+    {
+        string testString = "Hello";
+        string expectedExceptionMessage = "The value 'TransportWithMessageCredential' is not supported in this context for the binding security property 'securityMode'.";
+        ChannelFactory<IWcfService> factory = null;
+        IWcfService serviceProxy = null;
+
+        // BasicHttpsSecurityMode.TransportWithMessageCredential is accessible but not supported.
+        // Verify the correct exception and message is thrown.
+        // When/if Message Security is supported this test will fail and will serve as a reminder to add test coverage.
+        PlatformNotSupportedException exception = Assert.Throws<PlatformNotSupportedException>(() =>
+        {
+            try
+            {
+                // *** SETUP *** \\
+                NetHttpsBinding netHttpsBinding = new NetHttpsBinding(BasicHttpsSecurityMode.TransportWithMessageCredential);
+                factory = new ChannelFactory<IWcfService>(netHttpsBinding, new EndpointAddress(Endpoints.HttpBaseAddress_NetHttps));
+                serviceProxy = factory.CreateChannel();
+
+                // *** EXECUTE *** \\
+                string result = serviceProxy.Echo(testString);
+            }
+            finally
+            {
+                // *** ENSURE CLEANUP *** \\
+                ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
+            }
+        });
+
+        // *** ADDITIONAL VALIDATION *** \\
+        Assert.Equal(expectedExceptionMessage, exception.Message);
```

Comment:
This will probably fail in NET Native (retail) because the string resources are stripped.  I think we can probably live without the extra string validation for these kind of NotSupported tests.